packages/medusa/src/api/middlewares/await-middleware.ts

```diff
@@ -3,7 +3,7 @@ import { NextFunction, Request, RequestHandler, Response } from "express"
 type handler = (req: Request, res: Response) => Promise<void>
 
 export default (fn: handler): RequestHandler => {
-  return (req: Request, res: Response, next: NextFunction) => {
+  return async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
todo: We have two places where this function exists, in medusa and utils. Should we deprecate the one from medusa and use the one from utils. Also, in the mean time both needs to be updated

Reply:
We should prob. keep the one in utils, if possible. Will make this PR breaking

Reply:
I'll update the one in utils to match, and deprecate the one in Medusa (not deleting it) with the message: `@deprecated use `import { wrapHandler } from "@medusajs/utils"`. And then we can delete it from Medusa at some point when we are comfortable with making the breaking change.

Reply:
Yes I agree with that the approach @olivermrbl also, we have a lot of places where deprecated and new are mixed up, same for config etc. so a cleanup starting with the middleware is fine for me :)